src/python/pants/option/global_options.py

```diff
@@ -728,6 +728,16 @@ def register_bootstrap_options(cls, register):
             help="Whether to allow import statements in BUILD files",
         )
 
+        register(
+            "--build-file-prelude-globs",
+            advanced=True,
+            type=list,
+            default=[],
+            help="Python files to evaluate and whose symbols should be exposed to all BUILD files ."
+            "This allows for writing functions which create multiple rules, or set default "
+            "arguments for rules.",
```

Comment:
Totally fine if a followup, but it would be awesome if this pointed to a real world example. 

Have you seen the docs rewrite at https://pants.readme.io/docs? One of my next priorities is writing about "How to write plugins", and this would be a fantastic page :) 

I'd be happy to invite you as an editor if you'd like, or I can pull things like if you put an example in the PR description.